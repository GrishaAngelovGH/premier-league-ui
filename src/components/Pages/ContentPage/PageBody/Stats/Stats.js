import PropTypes from 'prop-types'

const Stats = ({ statsItems }) => (
    <div className='row no-gutters justify-content-center mt-3'>
        <div className='col-9 col-lg-6'>
            <div className='row text-secondary border-bottom'>
                <div className='col-9 col-md-10'>Players</div>
                <div className='col-2 col-md-2'>Goals</div>
            </div>

            {
                statsItems.map((v, i) => (
                    <div key={i} className='row no-gutters border-bottom'>
                        <div className='col-1 col-md-1 m-1'>
                            {v.position}
                        </div>
                        <div className='col-9 col-md-9'>
                            <div className='row m-1'>
                                {v.playerName}
                            </div>
                            <div className='row mb-1'>
                                <div className='col-1 col-md-1'>
                                    <img src={v.teamImage} width={25} />
                                </div>
                                <div className='col-7 col-md-5 text-secondary'>
                                    {v.team}
                                </div>
                            </div>
                        </div>

                        <div className='col-1 col-md-1 d-flex justify-content-end mt-1'>
                            {v.goals}
                        </div>
                    </div>
                ))
            }
        </div>
    </div>
)

Stats.propTypes = {
    statsItems: PropTypes.array.isRequired
}

export default Stats