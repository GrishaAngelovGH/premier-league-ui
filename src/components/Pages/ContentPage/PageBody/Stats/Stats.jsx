const Stats = ({ statsItems }) => (
	<div className='flex justify-center mt-3'>
		<div className='w-1/2'>
			<div className='flex text-gray-500 border-b'>
				<div className='w-5/6'>Players</div>
				<div className='w-1/6 text-end'>Goals</div>
			</div>

			{
				statsItems.map((v, i) => (
					<div key={i} className="flex items-center border-b border-gray-400 p-2">
						<div className="w-5/6">
							<div className="flex gap-2 items-center">
								<div>{v.position}</div>
								<div className="flex items-center gap-2">
									<img src={v.teamImage} className="w-[30px] h-[30px]" />
									<div>
										<div>{v.playerName}</div>
										<div className="text-gray-500">{v.team}</div>
									</div>
								</div>
							</div>
						</div>
						<div className="w-1/6 text-end">{v.goals}</div>
					</div>
				))
			}
		</div>
	</div>
)

export default Stats