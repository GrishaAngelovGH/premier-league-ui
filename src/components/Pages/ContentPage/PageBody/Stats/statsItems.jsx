import images from 'images'

export default [
	{ position: 1, playerName: 'Dominic Calvert-Lewin', teamImage: images['everton'], team: 'Eveton', goals: 10 },
	{ position: 2, playerName: 'Son Heung-min', teamImage: images['tottenham'], team: 'Tottenham', goals: 9 },
	{ position: 3, playerName: 'Jamie Vardy', teamImage: images['leicester'], team: 'Leicester City', goals: 8 },
	{ position: 4, playerName: 'Mohamed Salah', teamImage: images['liverpool'], team: 'Liverpool', goals: 8 },
	{ position: 5, playerName: 'Bruno Fernandes', teamImage: images['manUtd'], team: 'Man United', goals: 7 },
	{ position: 6, playerName: 'Callum Wilson', teamImage: images['newcastle'], team: 'Newcastle', goals: 7 },
	{ position: 7, playerName: 'Harry Kane', teamImage: images['tottenham'], team: 'Tottenham', goals: 7 },
	{ position: 8, playerName: 'Patrick Bamford', teamImage: images['leeds'], team: 'Leeds United', goals: 7 }
]

