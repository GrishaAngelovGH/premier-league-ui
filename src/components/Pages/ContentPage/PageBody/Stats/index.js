import Stats from './Stats'
import statsItems from './statsItems'

export { statsItems }

export default Stats