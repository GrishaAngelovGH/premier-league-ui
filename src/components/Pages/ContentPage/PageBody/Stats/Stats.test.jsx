import { render } from '@testing-library/react'

import Stats from './Stats'

test('should render Stats component', () => {
    const statsItems = [
        { position: 1, playerName: 'Dominic Calvert-Lewin', teamImage: 'img1', team: 'Eveton', goals: 10 },
    ]

    const view = render(<Stats statsItems={statsItems} />)

    expect(view).toMatchSnapshot()
})