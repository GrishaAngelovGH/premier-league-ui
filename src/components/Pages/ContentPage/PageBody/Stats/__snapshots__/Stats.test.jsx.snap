// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render Stats component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex justify-center mt-3"
      >
        <div
          class="w-1/2"
        >
          <div
            class="flex text-gray-500 border-b"
          >
            <div
              class="w-5/6"
            >
              Players
            </div>
            <div
              class="w-1/6 text-end"
            >
              Goals
            </div>
          </div>
          <div
            class="flex items-center border-b border-gray-400 p-2"
          >
            <div
              class="w-5/6"
            >
              <div
                class="flex gap-2 items-center"
              >
                <div>
                  1
                </div>
                <div
                  class="flex items-center gap-2"
                >
                  <img
                    class="w-[30px] h-[30px]"
                    src="img1"
                  />
                  <div>
                    <div>
                      Dominic Calvert-Lewin
                    </div>
                    <div
                      class="text-gray-500"
                    >
                      Eveton
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="w-1/6 text-end"
            >
              10
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex justify-center mt-3"
    >
      <div
        class="w-1/2"
      >
        <div
          class="flex text-gray-500 border-b"
        >
          <div
            class="w-5/6"
          >
            Players
          </div>
          <div
            class="w-1/6 text-end"
          >
            Goals
          </div>
        </div>
        <div
          class="flex items-center border-b border-gray-400 p-2"
        >
          <div
            class="w-5/6"
          >
            <div
              class="flex gap-2 items-center"
            >
              <div>
                1
              </div>
              <div
                class="flex items-center gap-2"
              >
                <img
                  class="w-[30px] h-[30px]"
                  src="img1"
                />
                <div>
                  <div>
                    Dominic Calvert-Lewin
                  </div>
                  <div
                    class="text-gray-500"
                  >
                    Eveton
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="w-1/6 text-end"
          >
            10
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
